www/common/drive-ui.js

```diff
@@ -4551,178 +4590,211 @@ define([
+                        if (pathToCheck.length > 0) {
+                            var checkRoot = manager.find([ROOT]);
+                            var firstKey = pathToCheck[0];
+                            if (checkRoot && checkRoot[firstKey]) {
+                                var sfId = manager.isSharedFolder(checkRoot[firstKey]) && checkRoot[firstKey];
+                                if (sfId) {
+                                    if (!manager.folders[sfId]) {
+                                        UI.warn(Messages.fm_deletedFolder);
+                                        return;
+                                    }
+                                    if (files.restrictedFolders[sfId]) {
+                                        UI.warn(Messages.fm_restricted);
+                                        return;
+                                    }
+                                }
+                            }
+                        }
+                        APP.displayDirectory(clickedPath);
+                    },
+                    onContextMenu: function (clickedPath, event) {
+                        openContextMenu('tree')(event);
+                    },
+                    onFolderExpanded: function (clickedPath, isOpen) {
+                        LS.setFolderOpened(clickedPath, isOpen);
+                        if (clickedPath.length === 1) { return; }
+                        // When collapsing a folder, close all its children
+                        if (!isOpen) {
+                            LS.removeFoldersOpened(clickedPath);
+                        }
+                    }
+                },
+                openFolders: filteredOpenFolders
+            });
+            
+            var $treeElement = $(treeElement);
+            // Drive specific features
+            $treeElement.find('.cp-app-drive-element-row').each(function() {
+                var $row = $(this);
+                var elPath = $row.data('path');
+                if (!elPath || elPath.length === 0) { return; }
+                var pathToCheck = elPath[0] === 'root' ? elPath : [ROOT].concat(elPath);
+                var isSharedFolderPath = pathToCheck.length > 1 && pathToCheck[pathToCheck.length - 1] === ROOT;
+                $row.find('.cp-app-drive-icon-folder').css("color", isSharedFolderPath ? getFolderColor(pathToCheck.slice(0, -1)) : getFolderColor(pathToCheck));
+
+                var checkRoot = manager.find([ROOT]);
+                var sfId = null;
+                var f = null;
+                var droppable = true;
+                
+                if (elPath.length > 1 && elPath[0] === 'root') {
+                    var firstKey = elPath[1];
+                    if (checkRoot && checkRoot[firstKey]) {
+                        sfId = manager.isSharedFolder(checkRoot[firstKey]) && checkRoot[firstKey];
+                        if (sfId) {
+                            f = folders[sfId];
```

Comment:
Same as a previous comment, this is not a way to check if an element is a shared folder or to get a shared folder id.
When you only have a path, you can use `var sfId = manager.isInSharedFolder(path);`